src/kernel/src/rtld/mod.rs

```diff
@@ -431,7 +432,29 @@ impl<E: ExecutionEngine> RuntimeLinker<E> {
         Ok(SysOut::ZERO)
     }
 
-    fn sys_dynlib_do_copy_relocations(self: &Arc<Self>, _i: &SysIn) -> Result<SysOut, SysErr> {
+    fn sys_dynlib_load_prx(self: &Arc<Self>, i: &SysIn) -> Result<SysOut, SysErr> {
+        let libname = unsafe { i.args[0].to_str(1024) }?.unwrap();
+        let _args: usize = i.args[1].into();
+        let p_id: *mut u32 = i.args[2].into();
+
+        if self.app.file_info().is_none() {
+            return Err(SysErr::Raw(EPERM));
+        }
+
+        //TODO implement the rest of this function
```

Comment:
Why skipping this part?

Reply:
I'm not really sure If I can figure the rest out correctly. This works to a point where we don't immediately crash.

Reply:
Okay. Try your best and I'll take a look at this later.